docker/daemon/daemon_dest.go

```diff
@@ -55,13 +46,30 @@ func newImageDestination(systemCtx *types.SystemContext, ref daemonReference) (t
 	ctx, goroutineCancel := context.WithCancel(context.Background())
 	go imageLoadGoroutine(ctx, c, reader, statusChannel)
 
+	// For github.com/docker/docker consumers, this works just as well as
+	//   refString := ref.String()
+	// because when reading the RepoTags strings, github.com/docker/docker/reference
+	// normalizes both of them to the same value.
+	//
+	// Doing it this way to include the normalized-out `docker.io[/library]` does make
+	// a difference for github.com/projectatomic/docker consumers, with the
+	// “Add --add-registry and --block-registry options to docker daemon” patch.
+	// These consumers treat reference strings which include a hostname and reference
+	// strings without a hostname differently.
+	//
+	// Using the host name here is more explicit about the intent, and it has the same
+	// effect as (docker pull) in projectatomic/docker, which tags the result using
+	// a hostname-qualified reference.
+	// See https://github.com/containers/image/issues/72 for a more detailed
+	// analysis and explanation.
+	refString := fmt.Sprintf("%s:%s", namedTaggedRef.FullName(), namedTaggedRef.Tag())
```

Comment:
This code is wrong for `docker-archive:` transports, because the filename of a `.tar` archive is not related to what `RepoTags` you're going to export. There might be a way around it, but for now this code doesn't _really_ belong in `tarfile.Transport`.